### ISSUE TITLE
Front-end highlighter redesign: Front end highlighter missing dialog role and properties

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1618]
Example code:
```
<div id="edac-highlight-panel-controls" class="edac-highlight-panel-controls" tabindex="0" style="display: flex; width: 400px; position: fixed; inset: 516px auto auto 687px;">
[...]
<div class="edac-highlight-panel-controls-title"><span class="edac-highlight-panel-controls-title-icon" aria-hidden="true"></span>Accessibility Checker</div>
[...]
</div>
```

Screenshots:

<img width="1697" height="975" alt="HTML of container in dev tools" src="https://github.com/user-attachments/assets/c1915519-0320-4370-89cd-e47006404d05" />

Screen reader announcement when dialog is first open - VoiceOver only announces the close button as it receives focus with no context as to where you are:
<img width="1117" height="521" alt="VoiceOver screenshot" src="https://github.com/user-attachments/assets/63e01afc-8c5e-47aa-8888-221b216a51ac" />

How to fix:
1. Add `role="dialog"` to `#edac-highlight-panel-controls`.
2. Add an id to `.edac-highlight-panel-controls-title`
3. Add `aria-labelledby` to `#edac-highlight-panel-controls` referencing the id of the title.
